plugins/destination/plugin.go

```diff
@@ -157,14 +150,13 @@ func (p *Plugin) Metrics() Metrics {
 	case unmanaged:
 		return p.client.Metrics()
 	case managed:
-		metrics := Metrics{}
-		p.metricsLock.RLock()
-		for _, m := range p.metrics {
-			metrics.Errors += m.Errors
-			metrics.Writes += m.Writes
+		metrics := new(Metrics)
+		p.RLock()
```

Comment:
Shouldn't this be a write lock, because we're updating the metric values?

Reply:
No, as we're only reading them. The only one we're updating is the local `metrics` variable

Reply:
Ah okay, I see, thanks!